app/src/store/data.ts

```diff
@@ -37,9 +37,10 @@ import {
 } from 'src/utils/chains';
 import { DefaultStorage, getStorageKey, setStorageKey } from 'src/utils/data/utils';
 import { GRANT_ROUND_ABI, ERC20_ABI } from 'src/utils/constants';
```

Comment:
(arbitrary comment location to start a discussion)

> Instead of attempting to close this gap, we will now only use RPC calls to fetch initial data as a fallback for if/when the graph fails (such as when there is no SUBGRAPH_URL for the chain). This requires additional testing - we still want to make sure the graph hasn't fallen behind, but we likely only need to introduce a constant buffer rather than starting from the last event the graph recorded (as this is unlikely to correlate with the last block the graph saw).

Do you mind expanding on this? I want to make sure I understand what tradeoffs were introduced and what (if anything) needs to be moved into a separate issue/PR to handle in the future. Is this correct:
- If we have a subgraph URL defined in `chains.ts`, we're all set and no further testing is needed
- If we do not have a subgraph URL  defined in `chains.ts`, things may not work as expected and this needs to be handled in a separate PR? (if so, I think that's fine and requiring a subgraph is ok for now)

Also not sure I totally follow the "constant buffer" aspect!

Reply:
Did what we spoke about earlier make sense?

--

Just to clear up whats written above:

- If we DON'T have a URL defined in `chains.ts` then we're all good, the RPC calls will work as normal and provided that we're not polling polygon, we shouldn't run in to any problems for quite a while - at which point we can simply deploy a subgraph for that chain. 

- If we DO have a URL defined in `chains.ts`, then we should look to make an RPC call for the last _n_ blocks after getting the state from the graph. This will ensure that our local storage (indexedDB) is up to do date with the chain rather than just the graph to cover the situation where the graph is behind. 

If we don't do what I describe above, then as we start listening for blocks and updating the local storage after the initial fetch from the graph, we will be updating the `latestBlock` to the chains latestBlock and not the graphs, so if the graph is behind for whatever reason, the user will have gaps in what they have cached. 

This is what I meant by a constant buffer - we should check the last _n_ blocks after fetching from the graph to avoid any gaps in the users cached state.